clang/lib/3C/RewriteUtils.cpp

```diff
@@ -229,8 +229,9 @@ static void emit(Rewriter &R, ASTContext &C) {
       };
 
       // Check whether we are allowed to write this file.
+      std::string ToConv = FE->tryGetRealPathName().str();
       std::string FeAbsS = "";
-      getCanonicalFilePath(std::string(FE->getName()), FeAbsS);
+      getCanonicalFilePath(ToConv, FeAbsS);
```

Comment:
I thought it would be good to at least have a comment about why we do this, and I ended up also drafting a change so that failure here reports an error diagnostic rather than asserting (though I guess we can debate whether the risk of failure is too remote to justify the extra code for the diagnostic).  Please consider [this change](https://github.com/correctcomputation/checkedc-clang/compare/one_clang_pass...one_clang_pass.file-path-tweaks) for incorporation into this PR.  If any of it is controversial, feel free to skip it.

I re-ran the regression tests and also manually tested the new diagnostics by temporarily sabotaging the code (changing `ToConv` to a nonexistent or non-canonical path); I can't think of a reasonable way to write regression tests for the new diagnostics.

Reply:
Your change may be too much of a tangent to incorporate in this PR. Can you make it into a separate PR against this branch? If it's accepted first I'll FF before merging, or if it's later it's probably a simple rebase ('cause squash). Does that work?

Reply:
Everything in my change is dependent on this PR, so I can't submit it first.  Since this PR is making changes to file path canonicalization, I would prefer it if my comments documenting those changes were included in the PR.  But the change to diagnostics (and the associated removal of `getCanonicalFilePath`) could be done later or not at all if we decide we don't want it, and it's a good point that we shouldn't block this PR on that decision.  So, are you willing to just copy the comments, or would you like me to generate a revised commit?

Reply:
Kyle and I decided to discuss this further in #508.